src/js/controllers/lockSetup.js

```diff
@@ -73,39 +90,28 @@ angular.module('copayApp.controllers').controller('lockSetupController', functio
     var config = configService.getSync();
     var savedMethod = config.lock && config.lock.method;
 
-    if (!selectedMethod)
-      saveConfig();
-    else if (selectedMethod == 'fingerprint') {
-      if (savedMethod == 'pin') {
-        askForDisablePin(function(disablePin) {
-          if (disablePin) saveConfig('fingerprint');
-          else init();
-        });
-      } else saveConfig('fingerprint');
+    if (!selectedMethod) {
+      if (!savedMethod) return;
+      if (savedMethod == 'pin') $state.transitionTo('tabs.pin', {
```

Comment:
It dont follow how after transitioning to tab.pin, the new  (null) `selectedMethod` will be saved.

Also, disabling fingerprint should also ask for the fingerprint.